manifests/profile/www_lib/vhosts/publishing.pp

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::www_lib::vhosts::publishing
+#
+# Wordpress virtual hosts for Publishing
+#
+# @example
+#   include nebula::profile::www_lib::vhosts::publishing
+class nebula::profile::www_lib::vhosts::publishing (
+  String $docroot = '/www/www.publishing/web',
+) {
+  apache::vhost { 'www.publishing-http':
+    servername => 'www.publishing.umich.edu',
+    ssl        => false,
+    port       => 80,
+    docroot    => $docroot,
+    rewrites   => [
+      {
+        rewrite_rule => '^(.*)$ https://%{HTTP_HOST}$1 [L,NE,R]'
+      },
+    ],
+  }
+
+  # Note that this vhost listens on 443 but is HTTP, because HAProxy terminates
+  # SSL. We pass an ENV var to signal that the request is over HTTPS for app
+  # URL generation, etc.
+  apache::vhost { 'www.publishing-https':
+    servername      => 'https://www.publishing.umich.edu',
```

Comment:
This is the only servername I can see with the protocol baked in—is that on purpose?

Reply:
Yeah, this is a little bit odd, but I'm preserving how it was done before. I think different apps/versions respond differently to the server name (as delivered in the environment for mod_php) and the HTTPS env var we set. Belt and suspenders, I guess.

The effect here is to say to PHP/WordPress: you are on `https://...`, even though that's normally only conditioned on whether SSL is enabled or not. SSL is never enabled in this configuration because it is done at the load balancer.